x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/AzureOpenAiService.java

```diff
@@ -351,7 +351,9 @@ public static InferenceServiceConfiguration get() {
 
                 configurationMap.put(
                     RESOURCE_NAME,
-                    new SettingsConfiguration.Builder().setDescription("The name of your Azure OpenAI resource.")
+                    new SettingsConfiguration.Builder(EnumSet.of(TaskType.TEXT_EMBEDDING, TaskType.COMPLETION)).setDescription(
```

Comment:
nit: would it make sense to use the `supportedTaskTypes` defined above for the configurations that support all supported task types and just explicitly define it for the configurations that don't support all of them?

Reply:
Yeah, I'll make that change 👍